airbyte-integrations/connectors/destination-redis/src/main/java/io/airbyte/integrations/destination/redis/RedisSslUtil.java

```diff
@@ -0,0 +1,155 @@
+  public static final String PARAM_SSL = "ssl";
+
+  private static final String CLIENT_CERTIFICATE = "client.crt";
+  private static final String CLIENT_CA_CERTIFICATE = "client-ca.crt";
+  private static final String CLIENT_KEY = "client.key";
+  private static final String CLIENT_KEY_STORE = "client_key_store.p12";
+  private static final String KEY_STORE_TYPE = "PKCS12";
+  private static final String TRUST_STORE = "truststore.jks";
+  private static final String TRUST_PASSWORD = "truststore_pwd";
+  private static final String TRUST_TYPE = "JKS";
+
+  /**
+   * set javax.net.ssl.keyStore and javax.net.ssl.trustStore based on provided ca.crt, client.crt, client.kay
+   *
+   * @param redisSslMode json ssl mode config
+   */
+  public static void setupCertificates(final JsonNode redisSslMode) {
+    try {
+      if (getSslVerifyMode(redisSslMode) == VERIFY_IDENTITY) {
+        LOGGER.info("Preparing ssl certificates for {} mode", PARAM_SSL_MODE_VERIFY_FULL);
+        final String clientKeyPassword = getOrGeneratePassword(redisSslMode);
+        initCertificateStores(redisSslMode.get(PARAM_CA_CERTIFICATE).asText(),
+            redisSslMode.get(PARAM_CLIENT_CERTIFICATE).asText(), redisSslMode.get(PARAM_CLIENT_KEY).asText(), clientKeyPassword);
+      }
+    } catch (final IOException | InterruptedException e) {
+      throw new RuntimeException("Failed to import certificate into Java Keystore");
+    }
+  }
+
+  /**
+   * Generate random pass if key pass param is empty
+   *
+   * @param sslModeConfig json ssl mode config
+   * @return client key password
+   */
+  private static String getOrGeneratePassword(final JsonNode sslModeConfig) {
+    final String clientKeyPassword;
+    if (sslModeConfig.has(PARAM_CLIENT_KEY_PASSWORD) && !sslModeConfig.get(PARAM_CLIENT_KEY_PASSWORD).asText().isEmpty()) {
+      clientKeyPassword = sslModeConfig.get(PARAM_CLIENT_KEY_PASSWORD).asText();
+    } else {
+      clientKeyPassword = RandomStringUtils.randomAlphanumeric(10);
+    }
+    return clientKeyPassword;
+  }
+
+  /**
+   * The method generate certificates based on provided ca.crt, client.crt, client.kay.
+   * Generated keys
+   *
+   * @param caCertificate certificate to validate client certificate and key.
+   * @param clientCertificate The client certificate.
+   * @param clientKey The client key.
+   * @param clientKeyPassword The client key password.
+   */
+  private static void initCertificateStores(
+      final String caCertificate,
+      final String clientCertificate,
+      final String clientKey,
+      final String clientKeyPassword)
+      throws IOException, InterruptedException {
+
+    LOGGER.info("Try to generate {}", CLIENT_KEY_STORE);
+    createCertificateFile(CLIENT_CERTIFICATE, clientCertificate);
+    createCertificateFile(CLIENT_KEY, clientKey);
+    runProcess("openssl pkcs12 -export -in " + CLIENT_CERTIFICATE + " -inkey " + CLIENT_KEY + " -out " + CLIENT_KEY_STORE + " -passout pass:"
```

Comment:
How ridiculous would it be if this string were refactored to some `String.format`? Right now this is a little hard to read and it can be easy to miss a `"` if this needs to be touched

Something like 
```java
runProcess(String.format("openssl pkcs12 -export -in %s -inkey %s -out %s -passout pass: %s",
    CLIENT_CERTIFICATE, 
    CLIENT_KEY, 
    CLIENT_KEY_STORE, 
    clientKeyPassword), 
    Runtime.getRuntime());
```

Reply:
I was struggling to make the key generation work and was not thinking much about readability. Thank you for point me out. `String.format` looks much cleaner let me format it